WIP: For 1.8: Feature 119

shimono opened the pull request.

shimono committed:
add *.log to .gitignore

shimono committed:
fix #119

shimono committed:
fix #119

tochi-y commented on src/main/resources/js-lib/personium-dao.js:
```diff
@@ -650,6 +649,22 @@ _p.Webdav.prototype.put = function(param, contentType, data, etag) {
+    }
+};
+_p.Webdav.prototype.updateFile = function(fileName, contentType, data, etag) {
+    if (typeof param == 'string') {
```
`param` is not included in arguments.

shimono replied:
fixed

tochi-y reviewed (changes requested):
I missed a operation. Request changes is correct.

shimono committed:
fix and improve the new methods davCol#createFile(), davCol#updateFile()

shimono committed:
fix and improve the new methods davCol#createFile(), davCol#updateFile()

shimono committed:
minor comment change

shimono committed:
Merge branch 'feature-119' of https://github.com/shimono/personium-engine into feature-119

shimono committed:
keep the compatibility for now and prepare for near future release (__put will replace current put)

shimono committed:
add new Method DavCollection#createOrUpdate()

shimono committed:
method name update (createOrUpdate -> updateOrCreateFile)

shimono committed:
Let createFile, updataFile only accept param json.
put should accept both if-match and if-none-match headers in the future, so renamed the param key of __put from etag to ifMatch for its preparation.
add comments.
